src/v/iceberg/values_json.cc

```diff
@@ -0,0 +1,674 @@
+};
+
+void value_to_json(
+  iceberg::json_writer&,
+  const iceberg::primitive_value&,
+  const iceberg::primitive_type&);
+void value_to_json(
+  iceberg::json_writer&,
+  const iceberg::struct_value&,
+  const iceberg::struct_type&);
+void value_to_json(
+  iceberg::json_writer&, const iceberg::list_value&, const iceberg::list_type&);
+void value_to_json(
+  iceberg::json_writer&, const iceberg::map_value&, const iceberg::map_type&);
+
+struct rjson_visitor {
+    explicit rjson_visitor(iceberg::json_writer& w)
+      : w(w) {}
+    void
+    operator()(const iceberg::boolean_value& v, const iceberg::boolean_type&) {
+        w.Bool(v.val);
+    }
+    void operator()(const iceberg::int_value& v, const iceberg::int_type&) {
+        w.Int(v.val);
+    }
+    void operator()(const iceberg::long_value& v, const iceberg::long_type&) {
+        w.Int64(v.val);
+    }
+    void operator()(const iceberg::float_value& v, const iceberg::float_type&) {
+        w.Double(v.val);
+    }
+    void
+    operator()(const iceberg::double_value& v, const iceberg::double_type&) {
+        w.Double(v.val);
+    }
+
+    void operator()(const iceberg::date_value& v, const iceberg::date_type&) {
+        const std::chrono::system_clock::time_point tp{
+          std::chrono::days(v.val)};
+        const std::chrono::year_month_day ymd{
+          std::chrono::floor<std::chrono::days>(tp)};
+        w.String(fmt::to_string(ymd));
+    }
+
+    void operator()(const iceberg::time_value& v, const iceberg::time_type&) {
+        using namespace std::chrono_literals;
+        const std::chrono::microseconds us{v.val};
+        const auto s = std::chrono::floor<std::chrono::seconds>(us);
+        const auto rest = (us - s) / 1us;
+
+        const std::chrono::system_clock::time_point tp{s};
+        const auto tt = std::chrono::system_clock::to_time_t(tp);
+        const auto tm = *std::gmtime(&tt);
+        w.String(fmt::format("{}.{:06}", std::put_time(&tm, "%T"), rest));
+    }
+
+    void operator()(
+      const iceberg::timestamp_value& v, const iceberg::timestamp_type&) {
+        using namespace std::chrono_literals;
+        const std::chrono::microseconds us{v.val};
+        const auto s = std::chrono::floor<std::chrono::seconds>(us);
+        const auto rest = (us - s) / 1us;
+
+        const std::chrono::system_clock::time_point tp{s};
+        const auto tt = std::chrono::system_clock::to_time_t(tp);
+        const auto tm = *std::gmtime(&tt);
+        w.String(fmt::format("{}.{:06}", std::put_time(&tm, "%FT%T"), rest));
+    }
+    void operator()(
+      const iceberg::timestamptz_value& v, const iceberg::timestamptz_type&) {
+        // Stores ISO - 8601 standard timestamp with microsecond precision;
+        // must include a zone offset and it must be '+00:00'.
+        // NOTE(oren): so this means a real timestamp offset is not supported or
+        // what? I guess this is just for downstream code that expects to find
+        // UTC?
```

Comment:
cruft comment, but of interest from the [spec](https://iceberg.apache.org/spec/#primitive-types):

>Timestamp values with time zone represent a point in time: values are stored as UTC and do not retain a source time zone (2017-11-16 17:10:34 PST is stored/retrieved as 2017-11-17 01:10:34 UTC and these values are considered identical).